From 7018206ea45db2e8bdfeb67d33f3387c9678a407 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Thu, 16 Aug 2012 11:19:16 +0200
Subject: [PATCH] patch: no-eula

Squashed commit of the following:

commit d13f1ccfdf4c150cab91105e9b8542ecbb048a9b
Author: Nils Philippsen <nils@redhat.com>
Date:   Fri Nov 19 12:20:52 2010 +0100

    don't show EULA, mention bugzilla in about dialog (#504344)
---
 src/xsane-text.h |  2 ++
 src/xsane.c      | 16 ++++++----------
 src/xsane.h      |  3 +++
 3 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/src/xsane-text.h b/src/xsane-text.h
index fc6bbeb..ee4a222 100644
--- a/src/xsane-text.h
+++ b/src/xsane-text.h
@@ -230,6 +230,8 @@
                                           "This program is distributed in the hope that it will be useful, but\n" \
                                           "WITHOUT ANY WARRANTY; without even the implied warranty of\n" \
                                           "MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.\n")
+#define TEXT_MODIFIED_BLURB             _("This package is modified from the original version.\n" \
+                                          "Please contact your vendor or report problems at")
 #define TEXT_EMAIL_ADR					_("E-mail:")
 #define TEXT_HOMEPAGE					_("Homepage:")
 #define TEXT_FILE					_("File:")
diff --git a/src/xsane.c b/src/xsane.c
index 1c5d61d..8b24b0c 100644
--- a/src/xsane.c
+++ b/src/xsane.c
@@ -3533,10 +3533,13 @@ static void xsane_about_dialog(GtkWidget *widget, gpointer data)
   snprintf(buf, sizeof(buf), "XSane %s %s\n"
                              "%s %s\n"
                              "\n"
+                             "%s\n%s"
+                             "\n\n"
                              "%s %s\n"
                              "%s %s\n",
                              TEXT_VERSION, XSANE_VERSION,
                              XSANE_COPYRIGHT_SIGN, XSANE_COPYRIGHT_TXT,
+                             TEXT_MODIFIED_BLURB, XSANE_BUGTRACKER_URL,
                              TEXT_HOMEPAGE, XSANE_HOMEPAGE,
                              TEXT_EMAIL_ADR, XSANE_EMAIL_ADR);
 
@@ -5733,6 +5736,7 @@ static int xsane_init(int argc, char **argv)
 
         case 'v': /* --version */
           g_print("%s-%s %s %s\n", xsane.prog_name, XSANE_VERSION, XSANE_COPYRIGHT_SIGN, XSANE_COPYRIGHT_TXT);
+          g_print("\n%s\n%s\n\n", TEXT_MODIFIED_BLURB, XSANE_BUGTRACKER_URL);
           g_print("  %s %s\n", TEXT_EMAIL_ADR, XSANE_EMAIL_ADR);
           g_print("  %s %s\n", TEXT_PACKAGE, XSANE_PACKAGE_VERSION);
           g_print("  %s%d.%d.%d\n", TEXT_GTK_VERSION, GTK_MAJOR_VERSION, GTK_MINOR_VERSION, GTK_MICRO_VERSION);
@@ -5859,17 +5863,9 @@ static int xsane_init(int argc, char **argv)
   }
 
 
-  if (xsane_pref_restore()) /* restore preferences, returns TRUE if license is not accpted yet */
+  if (xsane_pref_restore()) /* restore preferences, returns TRUE if the version is different from the last run */
   {
-    if (xsane_display_eula(1)) /* show license and ask for accept/not accept */
-    {
-      DBG(DBG_info, "user did not accept eula, we abort\n");
-      return 1; /* User did not accept eula */
-    }
-    else /* User did accept eula */
-    {
-      xsane_pref_save();
-    }
+    xsane_pref_save();
   }
 
   xsane_pref_restore_media();
diff --git a/src/xsane.h b/src/xsane.h
index 3d8caaa..6c7568e 100644
--- a/src/xsane.h
+++ b/src/xsane.h
@@ -98,6 +98,9 @@
 #define XSANE_EMAIL_ADR		"Oliver.Rauch@xsane.org"
 #define XSANE_HOMEPAGE		"http://www.xsane.org"
 #define XSANE_COPYRIGHT_TXT	XSANE_DATE " " XSANE_COPYRIGHT
+#ifndef XSANE_BUGTRACKER_URL
+#define XSANE_BUGTRACKER_URL             "(no bug tracker configured)"
+#endif
 
 /* ---------------------------------------------------------------------------------------------------------------------- */
 
-- 
1.7.11.4

